.github/workflows/post-merge-push.yaml

```diff
@@ -0,0 +1,58 @@
+name: Build latest binaries
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  build-and-push:
+    concurrency:
+      group: build-${{ github.head_ref || github.run_id }}
+      cancel-in-progress: true
+    name: Build and Unit Test
+    env:
+      REGISTRY_REPO: oci://ghcr.io/k0rdent/kcm/charts-ci
+    runs-on: ubuntu-latest
+    outputs:
+      version: ${{ steps.vars.outputs.version }}
+    permissions:
+      packages: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with:
+          go-version-file: 'go.mod'
+          cache: true # default
+      - name: Verify all generated pieces are up-to-date
+        run: make generate-all && git add -N . && git diff --exit-code
+      - name: Set up Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Login to GHCR
+        uses: docker/login-action@v3.4.0
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Get outputs
+        id: vars
+        run: |
+          GIT_VERSION=$(git describe --tags --always)
+          echo "version=${GIT_VERSION:1}" >> $GITHUB_OUTPUT
+      - name: Build and push KCM controller image to the public repository
+        uses: docker/build-push-action@v6
+        with:
+          build-args: |
+            LD_FLAGS=-s -w -X github.com/K0rdent/kcm/internal/build.Version=${{ steps.vars.outputs.version }}
+          context: .
+          platforms: linux/amd64
+          tags: |
+            ghcr.io/k0rdent/kcm/controller-ci:${{ steps.vars.outputs.version }}
+          push: true
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+      - name: Prepare and push KCM template charts to the public repository
+        run: |
+          make kcm-chart-release
+          make helm-push
```

Comment:
Here we're hoping that the `VERSION` variable will be set in the same way in Makefile as it was set in the `vars` step. I would like to give up this hope and set VERSION env here to the `${{ steps.vars.outputs.version }}`

Also we're building `ghcr.io/k0rdent/kcm/controller-ci` image, but it looks like chart will be build with the default `ghcr.io/k0rdent/kcm/controller`, since we're only patching the tag version. Extra image name patch is needed here